mongo-statefulset/README.md

```diff
@@ -135,74 +136,199 @@ Once successfully connected to the database, execute the `rs.status()` command t
+			"configVersion" : 4
+		},
+		{
+			"_id" : 3,
+			"name" : "mongodbreplsetabc-3.mongodb-replset-internal-abc.mongodb-statefulset.svc.cluster.local:27017",
+			"health" : 1,
+			"state" : 2,
+			"stateStr" : "SECONDARY",
+			"uptime" : 896,
+			"optime" : {
+				"ts" : Timestamp(1507687663, 1),
+				"t" : NumberLong(1)
+			},
+			"optimeDurable" : {
+				"ts" : Timestamp(1507687663, 1),
+				"t" : NumberLong(1)
+			},
+			"optimeDate" : ISODate("2017-10-11T02:07:43Z"),
+			"optimeDurableDate" : ISODate("2017-10-11T02:07:43Z"),
+			"lastHeartbeat" : ISODate("2017-10-11T02:07:50.781Z"),
+			"lastHeartbeatRecv" : ISODate("2017-10-11T02:07:52.384Z"),
+			"pingMs" : NumberLong(0),
+			"syncingTo" : "mongodbreplsetabc-2.mongodb-replset-internal-abc.mongodb-statefulset.svc.cluster.local:27017",
+			"configVersion" : 4
 		}
 	],
 	"ok" : 1
 }
 ```
+4 instances (1 primary and 3 secondary) is shown above and depicts a healthy replica set
+
+### Confirm the health of the two shards through the shard router
+
+Check the logs for mongos, for example
+
+```
+$ oc logs mongos-2-t5zp8
+```
+
+Log into the mongos pod
+
+```
+$ oc rsh mongos-2-t5zp8
+```
+
+Run the same command from earlier. This time you will see a "mongos" prompt.
+
+```
+$ mongo admin -u admin -p $MONGODB_ADMIN_PASSWORD
+```
+
+Check the shard status with "sh.status()". Notice our two replica sets are used as shard members. However, we have no databases enabled for sharding.
+
+```
+mongos> sh.status()
+--- Sharding Status ---
+  sharding version: {
+	"_id" : 1,
+	"minCompatibleVersion" : 5,
+	"currentVersion" : 6,
+	"clusterId" : ObjectId("59dd7887f68c97671ce9b289")
+}
+  shards:
+	{  "_id" : "rs0",  "host" : "rs0/mongodbreplsetabc-0.mongodb-replset-internal-abc.mongodb-statefulset.svc.cluster.local:27017,mongodbreplsetabc-1.mongodb-replset-internal-abc.mongodb-statefulset.svc.cluster.local:27017,mongodbreplsetabc-2.mongodb-replset-internal-abc.mongodb-statefulset.svc.cluster.local:27017,mongodbreplsetabc-3.mongodb-replset-internal-abc.mongodb-statefulset.svc.cluster.local:27017",  "state" : 1 }
+	{  "_id" : "rs1",  "host" : "rs1/mongodbreplsetdef-0.mongodb-replset-internal-def.mongodb-statefulset.svc.cluster.local:27017,mongodbreplsetdef-1.mongodb-replset-internal-def.mongodb-statefulset.svc.cluster.local:27017,mongodbreplsetdef-2.mongodb-replset-internal-def.mongodb-statefulset.svc.cluster.local:27017,mongodbreplsetdef-3.mongodb-replset-internal-def.mongodb-statefulset.svc.cluster.local:27017",  "state" : 1 }
+  active mongoses:
+	"3.4.6" : 1
+ autosplit:
+	Currently enabled: yes
+  balancer:
+	Currently enabled:  yes
+	Currently running:  no
+		Balancer lock taken at Wed Oct 11 2017 01:48:55 GMT+0000 (UTC) by ConfigServer:Balancer
+	Failed balancer rounds in last 5 attempts:  0
+	Migration Results for the last 24 hours:
+		No recent migrations
+  databases:
+```
+
+### Setting up sharding
+
+Enable our example database for sharding:
+
+```
+mongos> sh.enableSharding('example')
```

Comment:
can we find some way to do this in a more automated fashion?

Reply:
Sure, we can put all this inline